packages/devtools-connect/src/connect.spec.ts

```diff
@@ -412,6 +412,30 @@ describe('devtools connect', function () {
       expect(result.client).to.equal(mClient);
     });
 
+    it('allows lookup to be configured', async function () {
+      const uri = 'localhost:27017';
+      const mClient = stubConstructor(FakeMongoClient);
+      const mClientType = sinon.stub().returns(mClient);
+      mClient.connect.onFirstCall().resolves(mClient);
+      const lookupSpy = sinon.spy();
+      await connectMongoClient(
+        uri,
+        { ...defaultOpts, useSystemCA: false, lookup: lookupSpy },
+        bus,
+        mClientType as any,
+      );
+
+      // get options passed to mongo client
+      const finalClientOptions = mClientType.getCalls()[0].args[1];
+      expect(finalClientOptions.lookup).to.not.equal(lookupSpy); // lookup is always wrapped
+      finalClientOptions.lookup('localhost', 27017, () => {}); // invoke it the way it would be by net/tls
+      expect(lookupSpy.getCalls()).to.have.lengthOf(1); // verify our input lookup was called instead of the dns package
```

Comment:
bit of an odd test but since we're mocking things we don't actually get to the point where the driver would cause lookup to be invoked, by calling it here I'm checking that our input is wrapped rather than used directly.